### ISSUE TITLE
Include header row on copy

### ISSUE THREAD

**Issue by EmilianoLam** [mui/mui-x#14178]
### Search keywords

header row copy

### Latest version

- [X] I have tested the latest version

### Summary

Hello!,
Is it possible to include the header when the user presses ctrl + c on selected rows ?
or maybe by pressing "alt + c".
I am using version 7.12.1.
Thanks!

**Comment by michelengelen** [comment]
Hey @EmilianoLam this is currently not possible OOTB, but you can hook into the copy event with the `onClipboardCopy` prop on the datagrid. Targeting the columns might be hard to do though, since all info you get in that callback is the copied data as a string, a MUI specific event object and the grid api.

We might want to include this feature though. WDYT @joserodolfofreitas ?

**Comment by joserodolfofreitas** [comment]
Yes, that's indeed an interesting feature for copying and pasting. We briefly discussed this ability at some point, but we resolved that, more often than not, users won't want the headers.

>  but you can hook into the copy event with the onClipboardCopy prop on the datagrid

We can consider a feature out-of-the-box, but the above sounds like a good approach. It would be nice if we could have a recipe that's flexible to include the column headers based on a switch or other interactive element.

**Comment by EmilianoLam** [comment]
> Hey @EmilianoLam this is currently not possible OOTB, but you can hook into the copy event with the `onClipboardCopy` prop on the datagrid. Targeting the columns might be hard to do though, since all info you get in that callback is the copied data as a string, a MUI specific event object and the grid api.
> 
> We might want to include this feature though. WDYT @joserodolfofreitas ?

Hello there! I have manage to fulfill my needs this way. Thank you!

```
const onClipboardCopy = (data, event, { api }) => {
    const columns = api.getVisibleColumns();
    const columnsAsString = _
    .reduce(columns, (acc, col) => col.field !== '__check__' ? [...acc, col.headerName] : acc, [])
    .join('\t');

    navigator.clipboard.writeText(columnsAsString + '\n' + data);
  }
```

> Yes, that's indeed an interesting feature for copying and pasting. We briefly discussed this ability at some point, but we resolved that, more often than not, users won't want the headers.
> 
> > but you can hook into the copy event with the onClipboardCopy prop on the datagrid
> 
> We can consider a feature out-of-the-box, but the above sounds like a good approach. It would be nice if we could have a recipe that's flexible to include the column headers based on a switch or other interactive element.

Dear @joserodolfofreitas, as a small contribution to this new feature, I would say usability may not be hardly affected by using ALT + C instead of CTRL + C for anyone who wants to include headers on copy. 
Anyway, thanks for keeping this feature in mind.

**Comment by michelengelen** [comment]
As a middle ground I'll add the `waiting for 👍🏼` label here to see if the community wants that feature. From my point of view we should probably at least take a look into allowing to copy the headers, since that is currently not possible at all!
